src/ota/ota_handler.rs

```diff
@@ -473,6 +504,14 @@ mod tests {
 
         let mut state_mock = MockStateRepository::<PersistentState>::new();
         state_mock.expect_exists().returning(|| true);
+        ota.expect_get_primary()
+            .returning(|| Ok("rootfs.0".to_owned()));
```

Comment:
are we hard-coding a slot name here? If so, I'm not a big fan of this.

Reply:
Is only in the test side.